R/taxonomy.R

```diff
@@ -209,6 +219,42 @@ setMethod("checkTaxonomy", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+#' @rdname taxonomy-methods
+#'@export
+# Function to set taxonomy ranks
+setTaxonomyRanks <- function(rank) {
+    rank <- tolower(rank)
+    # Check if rank is a character vector with length >= 1
+    if (!is.character(rank) || length(rank) < 1 
+        || any(rank == "" | rank == " " | rank == "\t" | rank == "-" | rank == "_")
+        || any(grepl("\\s{2,}", rank))) {
+        stop("Input 'rank' should be a character vector with non-empty strings,
+             no spaces, tabs, hyphens, underscores, and non-continuous spaces."
+             , call. = FALSE)
+    }
+    .set_taxonomy_ranks(rank, detect.ranks = TRUE)
+    #assignInMyNamespace("TAXONOMY_RANKS", rank)
+}
+
+.set_taxonomy_ranks <- function(rank, detect.ranks = FALSE) {
```

Comment:
Maybe set.ranks instead of detect.ranks? --> The ranks are not "detected", they are just set based on colnames(rowData(x))...

This function could get TreeSE, ranks and set.ranks as input.
1. TreeSE: if ranks are not specified by user, get ranks from colnames(rowData(x))
2. ranks: User can specify ranks manually.
3. set.ranks: Set ranks based on user-specified ranks parameter or colnames(rowData(x))